Content.Server/Objectives/Systems/StealConditionSystem.cs

```diff
@@ -101,15 +103,18 @@ private float GetProgress(MindComponent mind, StealConditionComponent condition)
         //check stealAreas
         if (condition.CheckStealAreas)
         {
-            var areasQuery = AllEntityQuery<StealAreaComponent>();
-            while (areasQuery.MoveNext(out var uid, out var area))
+            var areasQuery = AllEntityQuery<TransformComponent, StealAreaComponent>();
+            while (areasQuery.MoveNext(out var uid, out var xform, out var area))
             {
                 if (!area.Owners.Contains(mind.Owner))
                     continue;
 
                 var nearestEnt = _lookup.GetEntitiesInRange(uid, area.Range);
                 foreach (var ent in nearestEnt)
                 {
+                    if (!_interaction.InRangeUnobstructed((uid, xform), ent, range: area.Range))
+                        continue;
+
```

Comment:
1. Use GetEntitiesInRange component version, it should already be returning contained entities with the query.
2. Use hashset cache.